contracts/ERC20WithPermit.sol

```diff
@@ -177,6 +166,15 @@ contract ERC20WithPermit is Ownable, IERC20WithPermit {
         _burn(account, amount);
     }
 
+    /* solhint-disable-next-line func-name-mixedcase */
+    function DOMAIN_SEPARATOR() public view override returns (bytes32) {
+        if (chainId() == cachedChainId) {
+            return cachedDomainSeparator;
+        } else {
+            return buildDomainSeparator();
```

Comment:
in case `chainId() =/= cachedChainId` we build a new domain separator. But do we also want to update cachedChainId, so the next time we don't need to build it again assuming chain id doesn't change that frequently..

Reply:
We don't because in practice, this situation should never happen and `cachedChainId` should be marked as `immutable` for gas efficiency which I forgot to do. So glad you brought it up! 

Please see 012346fdafebca91cc4504912c637d4f0daf701e

Reply:
okay, although I still remain a bit confused. Despite that this situation should never happen, yet we still have this check and updating cache within this validation seems reasonable. Anyway, I take that there is a veeery tiny chance it might happen and that's why we have it here. I'm ok leaving as-is.

Reply:
If we set the field as `immutable`, it will be cheaper to access it but also we cannot modify it later. Given that in practice we should never need to update it, we sacrifice gas efficiency updates but for the unlikely case.